src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -514,6 +558,52 @@ private static bool HasMatchingKey(IDictionary<string, string> secrets, string k
                 : ScriptSecretsType.Function;
         }
 
+        private void ValidateHostSecrets(HostSecretsInfo hostSecrets)
+        {
+            if (hostSecrets != null)
+            {
+                ValidateSecret(ScriptConstants.MasterKeyName, hostSecrets.MasterKey, SecretGenerator.MasterKeySeed);
+                ValidateSecrets(hostSecrets.FunctionKeys, SecretGenerator.FunctionKeySeed);
+                ValidateSecrets(hostSecrets.SystemKeys, SecretGenerator.SystemKeySeed);
+            }
+        }
+
+        private void ValidateSecret(string keyName, string keyValue, ulong seed, string functionName = null)
+        {
+            if (SecretGenerator.TryValidateSecret(keyValue, seed))
+            {
+                _metricsLogger.LogEvent(MetricEventNames.IdentifiableSecretLoaded, functionName);
+            }
+            else
+            {
+                _metricsLogger.LogEvent(MetricEventNames.NonIdentifiableSecretLoaded, functionName);
+
+                if (_strictHISFeatureEnabled.Value)
+                {
+                    // when strict mode is enabled, we won't let the application continue to run with
+                    // non-compliant secrets
+                    throw new InvalidOperationException(string.Format(Resources.NonHISSecret, SecretGenerator.GetKeyType(seed), keyName, functionName));
```

Comment:
I've decided to make this a hard error - the host will simply refuse to load a non-HIS secret in this mode. Depending on the codepath that is loading secrets, the behavior may differ. On an http request path (user http or admin APIs), the request will fail. On some host startup paths (e.g. when startup context cache is enabled [here](https://github.com/Azure/azure-functions-host/blob/ad0843d26bfe7c2555bcee250da1dacccdc61adf/src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs#L842)) the host startup will fail.